`test_trailing_slash_directory_metadata` fails on Python 3.15.0a8

## Description

Two unit tests fail on Python 3.15.0-alpha.8:

```
FAILED tests/unit/metadata/test_metadata.py::test_trailing_slash_directory_metadata[/path/to/foo.egg-info]
FAILED tests/unit/metadata/test_metadata.py::test_trailing_slash_directory_metadata[/path/to/foo.egg-info/]
```

```
importlib.metadata.MetadataNotFound: No package metadata was found.
```

Observed in CI: https://github.com/pypa/pip/actions/runs/24166791118/job/70530057240?pr=13900

## Root cause

`from_directory()` in `src/pip/_internal/metadata/importlib/_dists.py` L114-117 calls `importlib.metadata.Distribution.at()` with the given path. The test passes a non-existent path (`/path/to/foo.egg-info`) to verify trailing-slash handling. In Python 3.15 alpha, `importlib.metadata` now raises `MetadataNotFound` eagerly when accessing `.metadata` on a path with no actual metadata files, where older versions were lazy.

## Environment

- Python 3.15.0-alpha.8
- pip main branch
- Ubuntu (GitHub Actions runner)

--- a/news/13901.trivial.rst
+++ b/news/13901.trivial.rst
@@ -1,0 +1,2 @@
+Fixed ``test_trailing_slash_directory_metadata`` to use a real temporary
+directory with metadata, for compatibility with Python 3.15.

--- a/tests/unit/metadata/test_metadata.py
+++ b/tests/unit/metadata/test_metadata.py
@@ -135,15 +135,15 @@ def test_dist_found_in_zip(tmp_path: Path) -> None:
     assert Path(dist.location) == Path(location)
 
 
-@pytest.mark.parametrize(
-    "path",
-    [
-        "/path/to/foo.egg-info".replace("/", os.path.sep),
-        # Tests issue fixed by https://github.com/pypa/pip/pull/2530
-        "/path/to/foo.egg-info/".replace("/", os.path.sep),
-    ],
-)
-def test_trailing_slash_directory_metadata(path: str) -> None:
+@pytest.mark.parametrize("trailing_slash", [False, True])
+def test_trailing_slash_directory_metadata(
+    tmp_path: Path, trailing_slash: bool
+) -> None:
+    # Tests issue fixed by https://github.com/pypa/pip/pull/2530
+    egg_info = tmp_path / "foo.egg-info"
+    egg_info.mkdir()
+    (egg_info / "PKG-INFO").write_text("Metadata-Version: 1.0\nName: foo\n")
+    path = str(egg_info) + (os.sep if trailing_slash else "")
     dist = get_directory_distribution(path)
     assert dist.raw_name == dist.canonical_name == "foo"
-    assert dist.location == "/path/to".replace("/", os.path.sep)
+    assert dist.location == str(tmp_path)

